cmplxs/cmplxs.go

```diff
@@ -0,0 +1,954 @@
+	}
+	var norm float64
+	if L == 1 {
+		for _, val := range s {
+			norm += cmplx.Abs(val)
+		}
+		return norm
+	}
+	if math.IsInf(L, 1) {
+		for _, val := range s {
+			norm = math.Max(norm, cmplx.Abs(val))
+		}
+		return norm
+	}
+	for _, val := range s {
+		norm += math.Pow(cmplx.Abs(val), L)
+	}
+	return math.Pow(norm, 1/L)
+}
+
+// Polar generates cmplx.Polar, element-wise, for the elements of s, and returns slices.
+// phase is in radians.
+func Polar(r, p []float64, s []complex128) ([]float64, []float64) {
+	if len(r) != len(p) || len(r) != len(s) {
+		panic("cmplxs.Arg: length of the slices do not match")
+	}
+	for i, v := range s {
+		r[i], p[i] = cmplx.Polar(v)
+	}
+	return r, p
+}
+
+// Prod returns the product of the elements of the slice.
+// Returns 1 if len(s) = 0.
+func Prod(s []complex128) complex128 {
+	prod := complex(1.0, 0)
+	for _, val := range s {
+		prod *= val
+	}
+	return prod
+}
+
+// Rad converts degrees to radians for the elements of s and stores in s.
+func Rad(s []float64) {
+	for i, v := range s {
+		s[i] = v * math.Pi / 180
+	}
+}
+
+// RadTo converts degrees to radians for the elements of s and stores in dst.
+// dst is returned.
+// It panics if the lengths of dst and s are not equal.
+func RadTo(dst, s []float64) []float64 {
+	if len(dst) != len(s) {
+		panic("cmplxs.Real: length of s and dst do not match")
+	}
+	if len(s) == 0 {
+		return dst
+	}
+	for i, v := range s {
+		dst[i] = v * math.Pi / 180
+	}
+	return dst
+}
+
+// Real calculates real(s) and stores in dst. dst is returned.
+// It panics if the lengths of dst and s are not equal.
+func Real(dst []float64, s []complex128) []float64 {
+	if len(s) != len(dst) {
+		panic("cmplxs.Real: length of s and dst do not match")
+	}
+	if len(s) == 0 {
+		return dst
+	}
+	for i, v := range s {
+		dst[i] = real(v)
+	}
+	return dst
+}
+
+// Rect calculates cmplx.Rect(r, p) for the elements of r & p, and returns slice.
+// phase is in radians.
+func Rect(dst []complex128, r, p []float64) []complex128 {
+	if len(r) != len(p) || len(r) != len(dst) {
+		panic("cmplxs: Rect: dst, r & p slice lengths do not match")
+	}
+	for i, v := range r {
+		dst[i] = cmplx.Rect(v, p[i])
+	}
+	return dst
+}
+
+// Reverse reverses the order of elements in the slice.
+func Reverse(s []complex128) {
+	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
+		s[i], s[j] = s[j], s[i]
+	}
+}
+
+// ReverseTo reverses the order of elements in s and stores in dst.
+// dst is returned.
+// It panics if the lengths of dst and s are not equal.
+func ReverseTo(dst, s []complex128) []complex128 {
+	if len(s) != len(dst) {
+		panic("cmplxs.ReverseTo: length of s and dst do not match")
+	}
+	for i := 0; i < len(s); i++ {
+		dst[i] = s[len(s)-1-i]
+	}
+	return dst
+}
+
+// Round returns the half away from zero rounded value of x with prec precision.
+//
+// Special cases are:
+// 	Round(±0) = +0
+// 	Round(±Inf) = ±Inf
+// 	Round(NaN) = NaN
+func Round(x complex128, prec int) complex128 {
+	if x == 0 {
```

Comment:
This can just call `floats.Round` on the real and imaginary parts, after the conditional on zero. Then, if we ever get this correct, we get done in one place.